### PR TITLE
Mocha Unit Tests

### PR DESCRIPTION
**Pull request by success**

### CODE REVIEW

File: test/temple.js [hunk 1]
```diff
@@ -38,15 +38,19 @@ describe('Banner', function () {
...
   describe('#getTerms()', function () {
 
     before(async () => {
-      cache.terms = await banner.getTerms();
+      try {
+        cache.terms = await banner.getTerms();
+      } catch (e) {
+        console.error(e);
+      }
     });
```
Comment (studioph) [inline test/temple.js]: Using a try/catch here won't cause Mocha to report the test suite failing. We also shouldn't be using try/catch here anyways since the function isn't supposed to throw an error. The caching feature should probably integrated with the first unit test.

The same applies to the other unit tests in this file.

File: lib/index.js [hunk 2]
```diff
@@ -64,23 +64,59 @@ async function batchRequest(batchSize, pageSize, requestParams={}, method, schoo
...
+      // config[school].methods.forEach(method => {
+      //     this[methods[method].displayName] = banner[method]
+      // });
+    }
+    /**
+   *
+   * @param offset
+   * @param max
+   * @returns {Promise<any>}
+   */
```
Comment (studioph) [inline lib/index.js]: Why is this moved into the lib.index file? The class definition and functions are supposed to be separate. We only want to expose the methods for the given school through the constructor.

File: lib/index.js [hunk 3]
```diff
@@ -64,23 +64,59 @@ async function batchRequest(batchSize, pageSize, requestParams={}, method, schoo
...
+        return res.Body;
     },
-
+    /**
+     *
+     * @param term
+     * @param offset
+     * @param max
+     * @returns {Promise<any>}
+     */
```
Comment (studioph) [inline lib/index.js]: The JSDoc comments should ideally also have the type for each param
`@param term {number | string}` etc

File: lib/index.js [hunk 4]
```diff
@@ -91,49 +127,78 @@ module.exports = {
...
         let res = await bannerRequest(this.School, 'get_instructionalMethod');
-        return res.Data;
+        return res.Body;
     },
-
+    /**
+     *
+     * @param term
+     * @param crn
+     * @returns {Promise<any[]|BigUint64Array|Uint8ClampedArray|Uint32Array|Blob|Int16Array|Float64Array|SharedArrayBuffer|string|Uint16Array|ArrayBuffer|Int32Array|Float32Array|BigInt64Array|Uint8Array|Int8Array>}
```
Comment (studioph) [inline lib/index.js]: What's with all these return types? It should just be `string`

**Review by studioph — CHANGES_REQUESTED** [review]

File: lib/index.js [hunk 5]
```diff
@@ -67,10 +88,11 @@ async function batchRequest(batchSize, pageSize, requestParams={}, method, schoo
  * @class Class to interact with the Server
  */
 
-class Banner {
+class Banner { // Class makes it easier more readbale imo
```
Comment (studioph) [inline lib/index.js]: Yes, but the functions should not be directly in the class, otherwise you could call a method for which there is not request in Banner for that school, that is the entire purpose of the factory being used in the constructor. Please consider moving the class declaration and constructor back into their own file and importing the functions as was done before.

**Review by studioph — CHANGES_REQUESTED** [review]

File: lib/index.js (hunk 4 above)
Comment (success) [reply lib/index.js]: Resolved

**Comment by studioph** [comment]
Closing this as I will take over development from here
